gridcomps/History/MAPL_HistoryGridComp.F90

```diff
@@ -5070,13 +5083,41 @@ subroutine CopyStateItems(src, dst, rc)
           call ESMF_StateGet(src, itemNames(n), bundle(1), _RC)
           call ESMF_StateAdd(dst, bundle, _RC)
        end if
-    end do
+    end do 
 
     deallocate(itemTypes)
     deallocate(itemNames)
 
     _RETURN(ESMF_SUCCESS)
   end subroutine CopyStateItems
 
+  function get_acc_offset(current_time,ref_time,rc) result(acc_offset)
+     integer :: acc_offset
+     type(ESMF_Time), intent(in) :: current_time
+     integer, intent(in) :: ref_time
+     integer, optional, intent(out) :: rc
+
+     integer :: status
+     integer :: hour,minute,second,year,month,day,diff_sec
+     type(ESMF_Time) :: new_time
+     type(ESMF_TimeInterval) :: t_int
+
+     call ESMF_TimeGet(current_time,yy=year,mm=month,dd=day,h=hour,m=minute,s=second,_RC)
+     call MAPL_UnpackTime(ref_time,hour,minute,second)
+     call ESMF_TimeSet(new_time,yy=year,mm=month,dd=day,h=hour,m=minute,s=second,_RC)
+     t_int = new_time - current_time
+
+     call ESMF_TimeIntervalGet(t_int,s=diff_sec,_RC)
+     if (diff_sec == 0) then
+        acc_offset = 0
+     else if (diff_sec > 0) then
+        acc_offset = diff_sec - 86400
```

Comment:
Why we subtract only a single day?

Reply:
So the logic here is I have this acc_ref_time. I want to make an alarm that rights at that time in the future starting from the model start time.

Either is it 
1. the same as the model start time
2. in the future relative to the current time, so the alarm should ring that day
3. in the past, so the alarm should not ring until the next day.

Since all I can pass to the coupler this offset, for case 2, I actually need first ring time to be the PREVIOUS day, otherwise, the alarm won't ring during the first day. So that is what that is doing. A more elegant solution would be to pass the ref_times to the coupler varspec does not allow that now.

Reply:
My concern was if somehow the offset is more that 1 day. Probably not likely/impossible to occur